### DISCUSSION TITLE
How can I transform the value returned by a Controller?

### DISCUSSION THREAD

**Discussion by dandv** [react-hook-form/react-hook-form#1717] [General] [answered]
Is there a way to process the value returned by a Controller? For example, I have a component that must display two radio buttons, Yes and No, instead of a checkbox or switch, and I'd like to return a Boolean from that:

```ts
  return (
    <Controller
      as={
        <RadioGroup
          value={defaultValues[name]}
          {...otherProps}
        >
          <FormControlLabel value="yes" control={<Radio/>} label="Yes"/>
          <FormControlLabel value="no" control={<Radio/>} label="No"/>
        </RadioGroup>
      }
      name={name}
      control={control}
    />
  );
```

Or should this just be done in the submit handler?

**Comment by bluebill1049** [comment, marked as answer]
hey @dandv we are improving `Controller` in the next release: https://github.com/react-hook-form/react-hook-form/pull/1711

for now: take a look at this example: 

https://github.com/react-hook-form/react-hook-form/blob/master/examples/parseFormatInputValues.tsx

**Reply by drinzunza** [reply]
Moved to: https://github.com/react-hook-form/react-hook-form/blob/master/examples/V7/parseFormatInputValues.tsx

**Reply by ghost** [reply]
> Moved to: https://github.com/react-hook-form/react-hook-form/blob/master/examples/V7/parseFormatInputValues.tsx

Hello @drinzunza  do u have sample for input type radio?

**Reply by DanielBlytheCTO** [reply]
Hi, what happened to the `as` prop? I can't see it in the docs?
